src/types.h

```diff
@@ -252,9 +254,12 @@ typedef struct
         clone_info_t clone_info;
         struct
         {
-            u64 event_id;
             u32 buffer_length;
             file_info_t file_info;
+            // send the length of argv we are processing. This is
+            // necessary because unlike paths, argvs can have empty
+            // strings so we cannot rely on double null separators
+            u32 argv_len;
```

Comment:
This seems way more room than we need for the length; maybe a u16 length and u16 flags to store the truncated flag?

Reply:
`u16` for the truncated length makes sense. I think technically the total length can be bigger than that, although I think a `u32` will do it. The number of arguments is limited to 2^31-1, and each argument is limited to `PAGE_SIZE*32`: https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/binfmts.h#L16.

When actually doing the calculations then it also takes into account that the whole thing can't be more than 1/4 of stack size (defaulted to 8MB?) nor 3/4 or `_STK_LIM` (set to 8 MB). So if I am doing my math right, very very worse case combined argv + envp cannot be more than 6 MB, so less than 32 bits but more than 16 bits.

I'll change it with these values.

Reply:
Well, the kernel may accept ridiculously sized ARGV arrays, but we only have 32KB of room in the buffer we're copying to.  I don't know that there's much to be gained by knowing how much got truncated beyond 32KB (there may be a lot more, but is it of any value to us?), but it probably doesn't hurt to store it instead of just a flag.

Reply:
Changed it to use `u16` for the truncated length and then passing a flag saying it was truncated.